src/map/battle.c

```diff
@@ -1767,6 +1767,8 @@ static int battle_calc_skillratio(int attack_type, struct block_list *src, struc
 					skillratio += 100 + 100 * skill_lv;
 					if ((status_get_max_hp(target) / 100) <= 50)
 						skillratio *= 2;
+					if(sd && pc->checkskill(sd, SU_SPIRITOFLIFE))
```

Comment:
you can make it more clear.
 `if (sd != NULL && pc->checkskill(sd, SU_SPIRITOFLIFE) > 0)`

Reply:
Okay! I will update it !